010-anonymous-login.md

```diff
@@ -0,0 +1,114 @@
+
+- Anonymous user initiate the E-Mail verification process
+- Verifies his E-Mail via the link provided by the E-mail
+- Will only be converted to a normal user at this point.
+
+The process for OAuth2 should be very simple, once a user has linked an OAuth2 authentification to an anonymous account - it will be converted.
+
+For data consistency, the user should be provided the same ID at any point.
+
+### Endpoints
+
+#### `GET: /v1/account/sessions/anonymous`
+
+Allow a user to create and login to an anonymous user account.
+
+#### `POST: /v1/account/verification/anonymous`
+
+Allow a user to initiate converting their account to one that is authenticated via E-Mail. This will have the same functionality as the `POST/v1/account/verification` endpoint, except that an E-Mail address needs to be provided in the payload.
+
+#### `PUT: /v1/account/verification/anonymous`
+
+This will have the same functionality as the `PUT/v1/account/verification` endpoint to complete converting the anonymous account. Maybe one endpoint could serve both use cases, but I would prefer the single responsibility.
+
+### Prior art
+
+[prior-art]: #prior-art
+
+Luckily we have a lot of resources and examples around the internet that we can take as an example. I don't think I need to name them here, since a simple Google search will satisfy everyone's curiosity.
+
+<!--
+
+Discuss prior art, both the good and the bad, in relation to this proposal. A
+few examples of what this can include are:
+
+- Does this functionality exist in other software and what experience has their
+  community had?
+- For other teams: What lessons can we learn from what other communities have
+  done here?
+- Papers: Are there any published papers or great posts that discuss this? If
+  you have some relevant papers to refer to, this can serve as a more detailed
+  theoretical background.
+
+This section is intended to encourage you as an author to think about the
+lessons from other software, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us
+whether they are brand new or if it is an adaptation from other software.
+
+Write your answer below.
+-->
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+**What happens if the user wants to convert his anonymous account to his already existing user account?**
+
+To be fair, I have no idea how to handle this properly. We could convert his data to his old account and also transfer all his stuff to the old user id. But this would break consistency, if there are any references in the database to the anonymous user id.
```

Comment:
I would just leave this in the hands of app makers. If they wish, I guess our API flexibly could already allow that kind of "accounts merge" or with minimum intervention from our side.

Reply:
Okay 👍🏻